packages/discord.js/src/managers/GuildManager.js

```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
    */
   async create({
     name,
-    afkChannelId,
-    afkTimeout,
-    channels = [],
+    icon = null,
+    verificationLevel,
     defaultMessageNotifications,
     explicitContentFilter,
-    icon = null,
     roles = [],
+    channels = [],
+    afkChannelId,
+    afkTimeout,
     systemChannelId,
     systemChannelFlags,
-    verificationLevel,
   }) {
-    icon = await DataResolver.resolveImage(icon);
-
-    for (const channel of channels) {
-      channel.parent_id = channel.parentId;
-      delete channel.parentId;
-      channel.user_limit = channel.userLimit;
-      delete channel.userLimit;
-      channel.rate_limit_per_user = channel.rateLimitPerUser;
-      delete channel.rateLimitPerUser;
-      channel.rtc_region = channel.rtcRegion;
-      delete channel.rtcRegion;
-      channel.video_quality_mode = channel.videoQualityMode;
-      delete channel.videoQualityMode;
-
-      if (!channel.permissionOverwrites) continue;
-      for (const overwrite of channel.permissionOverwrites) {
-        overwrite.allow &&= PermissionsBitField.resolve(overwrite.allow).toString();
-        overwrite.deny &&= PermissionsBitField.resolve(overwrite.deny).toString();
-      }
-      channel.permission_overwrites = channel.permissionOverwrites;
-      delete channel.permissionOverwrites;
-    }
-    for (const role of roles) {
-      role.color &&= resolveColor(role.color);
-      role.permissions &&= PermissionsBitField.resolve(role.permissions).toString();
-    }
-    systemChannelFlags &&= SystemChannelFlagsBitField.resolve(systemChannelFlags);
-
     const data = await this.client.rest.post(Routes.guilds(), {
       body: {
         name,
-        icon,
+        icon: icon && (await DataResolver.resolveImage(icon)),
         verification_level: verificationLevel,
         default_message_notifications: defaultMessageNotifications,
         explicit_content_filter: explicitContentFilter,
-        roles,
-        channels,
+        roles: roles.map(({ color, permissions, ...options }) => ({
+          ...options,
+          color: color && resolveColor(color),
+          permissions: permissions && PermissionsBitField.resolve(permissions).toString(),
```

Comment:
I'm aware this is the same as before, but permissions won't resolve (and will instead fail) if it's set to the number `0` or the bigint `0n`.

This applies to all the other occurrences (roles, flags...).

Reply:
Should be done with 28d9dbaddf173ed3f7317d44faff2ee1fa93bcd8!